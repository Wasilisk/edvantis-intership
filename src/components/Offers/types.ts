import {OfferType} from "../../models";

export type OffersProps = {
    offers: OfferType[]
}