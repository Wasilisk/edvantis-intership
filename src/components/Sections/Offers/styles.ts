import styled from "styled-components";
import {OfferCardStyles} from "../../Cards/OfferCard/styles";

export const OffersStyles = styled.div`
  max-width: 1200px;
  margin: 0 auto;
  padding: 100px 24px 50px 24px;
  display: flex;
  justify-content: space-between;
  align-items: stretch;
  
  ${OfferCardStyles} {
    margin: 0 15px;
  }

  @media (max-width: 768px) {
    flex-direction: column;
    padding: 80px 24px 50px 24px;
    
    ${OfferCardStyles} {
      margin: 0;
    }
    ${OfferCardStyles}:first-child {
      margin-bottom: 50px;
    }
  }
`;