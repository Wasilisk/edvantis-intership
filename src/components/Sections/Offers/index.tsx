import React from 'react';
import { OffersStyles } from './styles';
import {OffersProps} from "./types";
import OfferCard from "../../Cards/OfferCard";

const Offers = ({offers}: OffersProps) => {
    return (
        <OffersStyles id="#collection">
            {
                offers.length >= 2 && <>
                    <OfferCard variant="dark" {...offers[0]}/>
                    <OfferCard variant="light" {...offers[1]}/>
                </>
            }
        </OffersStyles>
    );
};

export default Offers;