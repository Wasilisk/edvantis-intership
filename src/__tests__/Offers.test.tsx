import {render} from "./__utils__/store-wrapper";
import Offers from "../components/Sections/Offers";
import {mockOffers} from "./__mock__/mock-offers";

describe('Offers component', () => {
    it('should render component', () => {
        const {container, asFragment} = render(<Offers offers={mockOffers.slice(0,2)}/>)

        expect(container.firstChild!.childNodes.length).toEqual(2);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render component if offers more then 2', () => {
        const {container, asFragment} = render(<Offers offers={mockOffers}/>)

        expect(container.firstChild!.childNodes.length).toEqual(2);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should be empty if offers less then 2', () => {
        const {container, asFragment} = render(<Offers offers={mockOffers.slice(0,1)}/>)

        expect(container.firstChild).toBeEmptyDOMElement()
        expect(asFragment()).toMatchSnapshot();
    });
})