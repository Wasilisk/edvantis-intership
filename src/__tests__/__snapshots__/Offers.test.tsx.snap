// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers component should be empty if offers less then 2 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi bfslMT"
  />
</DocumentFragment>
`;

exports[`Offers component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi bfslMT"
  >
    <div
      class="sc-bczRLJ czKsmx"
    >
      <div
        class="text-info"
      >
        <h6
          class="caption"
        >
          Caption
        </h6>
        <h4
          class="title"
        >
          Title
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
      <a
        class="sc-hKMtZM bkACdt"
        href="/link"
      >
        <button
          class="sc-dkzDqf hJRqwe"
        >
          Link
          <svg
            height="100%"
            viewBox="0 0 36 36"
            width="100%"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              class="cls-1"
              d="M34,18l-5-4v3H26v2h3v3ZM24,19H20V17h4v2Zm-6,0H14V17h4v2Zm-6,0H8V17h4v2ZM4,16a2,2,0,1,1-2,2A2,2,0,0,1,4,16Z"
              data-name="28"
              fill="currentColor"
            />
          </svg>
        </button>
      </a>
    </div>
    <div
      class="sc-bczRLJ fcOWbv"
    >
      <div
        class="text-info"
      >
        <h6
          class="caption"
        >
          Caption
        </h6>
        <h4
          class="title"
        >
          Title
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
      <a
        class="sc-hKMtZM bkACdt"
        href="/link"
      >
        <button
          class="sc-dkzDqf kfXHQN"
        >
          Link
          <svg
            height="100%"
            viewBox="0 0 36 36"
            width="100%"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              class="cls-1"
              d="M34,18l-5-4v3H26v2h3v3ZM24,19H20V17h4v2Zm-6,0H14V17h4v2Zm-6,0H8V17h4v2ZM4,16a2,2,0,1,1-2,2A2,2,0,0,1,4,16Z"
              data-name="28"
              fill="currentColor"
            />
          </svg>
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Offers component should render component if offers more then 2 1`] = `
<DocumentFragment>
  <div
    class="sc-gsnTZi bfslMT"
  >
    <div
      class="sc-bczRLJ czKsmx"
    >
      <div
        class="text-info"
      >
        <h6
          class="caption"
        >
          Caption
        </h6>
        <h4
          class="title"
        >
          Title
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
      <a
        class="sc-hKMtZM bkACdt"
        href="/link"
      >
        <button
          class="sc-dkzDqf hJRqwe"
        >
          Link
          <svg
            height="100%"
            viewBox="0 0 36 36"
            width="100%"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              class="cls-1"
              d="M34,18l-5-4v3H26v2h3v3ZM24,19H20V17h4v2Zm-6,0H14V17h4v2Zm-6,0H8V17h4v2ZM4,16a2,2,0,1,1-2,2A2,2,0,0,1,4,16Z"
              data-name="28"
              fill="currentColor"
            />
          </svg>
        </button>
      </a>
    </div>
    <div
      class="sc-bczRLJ fcOWbv"
    >
      <div
        class="text-info"
      >
        <h6
          class="caption"
        >
          Caption
        </h6>
        <h4
          class="title"
        >
          Title
        </h4>
        <p
          class="description"
        >
          Description
        </p>
      </div>
      <a
        class="sc-hKMtZM bkACdt"
        href="/link"
      >
        <button
          class="sc-dkzDqf kfXHQN"
        >
          Link
          <svg
            height="100%"
            viewBox="0 0 36 36"
            width="100%"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              class="cls-1"
              d="M34,18l-5-4v3H26v2h3v3ZM24,19H20V17h4v2Zm-6,0H14V17h4v2Zm-6,0H8V17h4v2ZM4,16a2,2,0,1,1-2,2A2,2,0,0,1,4,16Z"
              data-name="28"
              fill="currentColor"
            />
          </svg>
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
